src/libYARP_pcl/src/CMakeLists.txt

```diff
@@ -20,9 +20,11 @@ target_compile_features(YARP_pcl INTERFACE cxx_std_14)
 target_link_libraries(YARP_pcl INTERFACE YARP::YARP_sig)
 list(APPEND YARP_pcl_PUBLIC_DEPS YARP_sig)
 
-target_include_directories(YARP_pcl SYSTEM INTERFACE ${PCL_INCLUDE_DIR})
-target_link_libraries(YARP_pcl INTERFACE ${PCL_LIBRARIES})
-list(APPEND YARP_pcl_PUBLIC_DEPS PCL)
+target_include_directories(YARP_pcl SYSTEM INTERFACE ${PCL_COMMON_INCLUDE_DIR}
+                                                     ${PCL_IO_INCLUDE_DIR})
+target_link_libraries(YARP_pcl INTERFACE ${PCL_COMMON_LIBRARIES}
+                                         ${PCL_IO_LIBRARIES})
```

Comment:
Similarly to the above comment, I'd write `${PCL_LIBRARIES}` even if dealing with PCL 1.9+ and CMake targets. The reason is that on PCL 1.10 neither `${PCL_XXX_LIBRARIES}` nor individual targets deal with 3rd party dependencies. So this is a similar issue to the include dirs counterpart, but here we'll probably see errors on link time.

Reply:
reverted see b0443ce491b6793746eb8fcb0a586b6e2d2f4b1f